palette/moreland/luminance.go

```diff
@@ -0,0 +1,233 @@
+func (l *Luminance) Min() float64 {
+	return 0
+}
+
+// Palette fulfils the palette.Palette interface.
+type Palette []color.Color
+
+// Colors fulfils the palette.Palette interface.
+func (p Palette) Colors() []color.Color {
+	return p
+}
+
+// Palette returns an object that fulfills the palette.Palette interface,
+// where nColors is the number of desired colors.
+func (l Luminance) Palette(nColors int) Palette {
+	if l.max == 0 {
+		l.max = 1
+	}
+	delta := l.max / float64(nColors-1)
+	v := 0.0
+	c := make([]color.Color, nColors)
+	for i := 0; i < nColors; i++ {
+		var err error
+		c[i], err = l.At(v)
+		if err != nil {
+			panic(err)
+		}
+		v += delta
+	}
+	return Palette(c)
+}
+
+// BlackBody is based on the colors of black body radiation.
+// Although the colors are inspired by the wavelengths of light from
+// black body radiation, the actual colors used are designed to be
+// perceptually uniform. Colors of the desired brightness and hue are chosen,
+// and then the colors are adjusted such that the luminance is perceptually
+// linear (according to the CIELAB color space).
+func BlackBody() *Luminance {
+	return &Luminance{
+		colors: []cieLAB{
+			cieLAB{L: 0, A: 0, B: 0},
+			cieLAB{L: 39.112572747719774, A: 55.92470934659227, B: 37.65159714510402},
+			cieLAB{L: 58.45705480680232, A: 43.34389690857626, B: 65.95409116544081},
+			cieLAB{L: 84.13253643355525, A: -6.459770854468639, B: 82.41994470228775},
+			cieLAB{L: 100, A: 0, B: 0}},
+		scalars: []float64{0, 0.39112572747719776, 0.5845705480680232, 0.8413253643355525, 1},
+		Alpha:   1}
+}
+
+// ExtendedBlackBody derives a color map based on the colors of black body radiation
+// with some blue and purple hues thrown in at the lower end to add some "color."
+// The color map is similar to the default colors used in gnuplot. Colors of
+// the desired brightness and hue are chosen, and then the colors are adjusted
+// such that the luminance is perceptually linear (according to the CIELAB
+// color space).
+func ExtendedBlackBody() *Luminance {
+	return &Luminance{
+		colors: []cieLAB{
+			cieLAB{L: 0, A: 0, B: 0},
+			cieLAB{L: 21.873483862751876, A: 50.19882295659109, B: -74.66982659778306},
+			cieLAB{L: 34.506542513775905, A: 75.41302687474061, B: -88.73807072507786},
+			cieLAB{L: 47.02980511087303, A: 70.93217189227919, B: 33.59880053746508},
+			cieLAB{L: 65.17482203230537, A: 49.14591409658836, B: 56.86480950937553},
+			cieLAB{L: 84.13253643355525, A: -6.459770854468639, B: 82.41994470228775},
+			cieLAB{L: 100, A: 0, B: 0},
+		},
+		scalars: []float64{0, 0.21873483862751875, 0.34506542513775906, 0.4702980511087303,
+			0.6517482203230537, 0.8413253643355525, 1},
+		Alpha: 1,
+	}
+}
+
+// Kindlmann uses the colors first proposed in a paper
+// by Kindlmann, Reinhard, and Creem. The map is basically the rainbow
+// color map with the luminance adjusted such that it monotonically
+// changes, making it much more perceptually viable.
```

Comment:
Copied from the author's website. I think it's referring to the fact that these thing are evaluated based on the ability of test subjects to perform value-differentiation tasks with different color maps, so "perceptually viable" would refer to how well the subjects perform with a given color map. But yes, it is awkward. Suggestions welcome.